tuning/utils/preprocessing_utils.py

```diff
@@ -0,0 +1,329 @@
+) -> Dataset:
+    """Loads the dataset and applies the tokenizer + custom masking logic.
+
+    Args:
+        data_path: str
+            Path to the file to be loaded.
+        tokenizer: AutoTokenizer
+            Loaded tokenizer object to be used by the collator.
+        max_sequence_length: int
+            Max sequence length to be used for sequence tokenization.
+        input_field_name: str
+            Name of the input field in the data.
+        output_field_name: str
+            Name of the output field in the data.
+
+    Returns:
+        Dataset
+            HF Dataset with the pretokenized data.
+    """
+    dataset = load_hf_dataset_from_jsonl_file(
+        data_path, input_field_name, output_field_name
+    )
+    return dataset.map(
+        preprocess_and_tokenize,
+        fn_kwargs={
+            "tokenizer": tokenizer,
+            "max_seq_length": max_sequence_length,
+            "input_field_name": input_field_name,
+            "output_field_name": output_field_name,
+        },
+        remove_columns=[input_field_name, output_field_name],
+    )
+
+
+### Utils for loading the data from disk in supported formats [currently only jsonl]
+def load_hf_dataset_from_jsonl_file(
+    data_path: str, input_field_name: str, output_field_name: str
+) -> Dataset:
+    """Loads the huggingface datase as a generator.
+
+    Args:
+        data_path: str
+            Path to the file to be loaded.
+        input_field_name: str
+            Name of the input field in the data.
+        output_field_name: str
+            Name of the output field in the data.
+
+    Returns:
+        Dataset
+            HF Dataset with the data to be tokenized.
+    """
+    if input_field_name == output_field_name:
+        raise ValueError("Input field name and output field name should not match!")
+
+    def get_jsonl_object():
+        with open(data_path, "r", encoding="utf-8") as jsonl_file:
+            data_stream = [json.loads(line) for line in jsonl_file]
+            for data in data_stream:
+                yield {
+                    input_field_name: data[input_field_name],
+                    output_field_name: data[output_field_name],
+                }
+
+    return Dataset.from_generator(get_jsonl_object)
+
+
+### Utils for custom masking / manipulating input / output strs, etc
+def combine_sequence(input_element: str, output_element: str):
+    """Combines / concatenates input & output element.
+
+    Args:
+        input_element: str
+            Input component of the combined sequence.
+        output_element: str
+            Output component of the combined sequence.
+
+    Returns:
+        str
+            Sequence combined with whitespace.
+    """
+    if not input_element.endswith((" ", "\n", "\t")) and not output_element.startswith(
+        (" ", "\n", "\t")
+    ):
+        return input_element + " " + output_element
+    return input_element + output_element
+
+
+def preprocess_and_tokenize(
```

Comment:
I think here we ll need to accept template as argument and use template to combine sequence